src/ogdc_runner/argo.py

```diff
@@ -22,71 +22,234 @@
 )
 
 
-def _configure_argo_settings() -> WorkflowsService:
-    """Configure argo settings for the OGDC and return an argo WorkflowsService instance.
+class ArgoConfig:
+    """Configuration for Argo workflows."""
 
-    Environment variables can be used to override common defaults:
+    def __init__(
+        self,
+        namespace: str,
+        service_account_name: str,
+        workflows_service_url: str,
+        runner_image: str,
+        runner_image_tag: str,
+        image_pull_policy: str,
+    ):
+        self._namespace = namespace
+        self._service_account_name = service_account_name
+        self._workflows_service_url = workflows_service_url
+        self._runner_image = runner_image
+        self._runner_image_tag = runner_image_tag
+        self._image_pull_policy = image_pull_policy
 
-    * ARGO_NAMESPACE: defines the kubernetes namespace the OGDC argo instance is deployed to.
-    * ARGO_SERVICE_ACCOUNT_NAME: defines the Argo service account with permissions to execute workflows.
-    * ARGO_WORKFLOWS_SERVICE_URL: the OGDC Argo workflows service URL.
-    * OGDC_RUNNER_IMAGE_TAG: the docker image tag to use for `ogdc-runner` in
-      non-development environments. Defaults to "latest".
-    * ENVIRONMENT: if set to `dev`, a local `ogdc-runner` image will be used
-      instead of pulling a version from GHCR. Build with `docker build -t
-      ogdc-runner .` in the `rancher-desktop` docker context.
+    @property
+    def namespace(self) -> str:
+        """Get the namespace."""
+        return self._namespace
 
-    Returns a `WorkflowsService` instance.
+    @namespace.setter
+    def namespace(self, value: str) -> None:
+        """Set the namespace."""
+        self._namespace = value
```

Comment:
Does `namespace` need this getter/setter, since it is a simple str and does not perform any validation or other logic? It seems much more straightforward to me to simply have a property at init called `self.namespace`:

```
class Foo():
    def __init__(self, name: str):
        self.name = name


class ArgoConfig:
    def __init__(self, namespace: str):
        self.namespace = namespace


config = ArgoConfig(namespace="foo")

print(config.namespace)
config.namespace = "bar"
print(config.namespace)
```
```
foo
bar
```